### PR TITLE
Preserve blocked pointers used by `tt.load` operation with DPAS layout

### LINKED ISSUE #2378: Failure to compile `gemm_postop_addmatrix_benchmark.py` with
The `gemm_postop_addmatrix_benchmark.py` benchmark fails to compile with configuration:
```
        triton.Config(
            {'BLOCK_SIZE_M': 256, 'BLOCK_SIZE_N': 256, 'BLOCK_SIZE_K': 32, 'GROUP_SIZE_M': 4, 'grf_mode': 'large'},
            num_stages=3, num_warps=32),
```
because the compiler allocates shared memory buffers that exceed the capacity. To reproduce:

USE_IPEX=0 python gemm_postop_addmatrix_benchmark.py          

```                                                                                                            
Traceback (most recent call last):
  File "/home/jovyan/intel-xpu-backend-for-triton/benchmarks/triton_kernels_benchmark/gemm_postop_addmatrix_benchmark.py", line 255, in <module>
    benchmark.run(show_plots=False, print_data=True)
  File "/home/jovyan/.conda/envs/triton-3.10/lib/python3.10/site-packages/triton_kernels_benchmark-0.0.0-py3.10.egg/triton_kernels_benchmark/benchmark_testing.py", line 373, in run
    result_dfs.append(self._run(bench, save_path, show_plots, print_data, **kwargs))
  File "/home/jovyan/.conda/envs/triton-3.10/lib/python3.10/site-packages/triton_kernels_benchmark-0.0.0-py3.10.egg/triton_kernels_benchmark/benchmark_testing.py", line 307, in _run
    ret = self.fn(**x_args, **{bench.line_arg: y}, **bench.args, **kwrags)
  File "/home/jovyan/intel-xpu-backend-for-triton/benchmarks/triton_kernels_benchmark/gemm_postop_addmatrix_benchmark.py", line 242, in benchmark
    benchmark_suit.assert_close(triton_fn(), torch_fn(), atol=1e-4, rtol=rtol, err_msg='triton to torch')
  File "/home/jovyan/intel-xpu-backend-for-triton/benchmarks/triton_kernels_benchmark/gemm_postop_addmatrix_benchmark.py", line 239, in <lambda>
    triton_fn = lambda: matmul(a, b, d, c)
  File "/home/jovyan/intel-xpu-backend-for-triton/benchmarks/triton_kernels_benchmark/gemm_postop_addmatrix_benchmark.py", line 188, in matmul
    matmul_kernel_with_block_pointers[grid](
  File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/runtime/jit.py", line 330, in <lambda>
    return lambda *args, **kwargs: self.run(grid=grid, warmup=False, *args, **kwargs)
  File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/runtime/autotuner.py", line 168, in run
    ret = self.fn.run(
  File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/runtime/jit.py", line 687, in run
    kernel.run(grid_0, grid_1, grid_2, stream, kernel.function, kernel.packed_metadata, launch_metadata,
  File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/compiler/compiler.py", line 417, in __getattribute__
    self._init_handles()
  File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/compiler/compiler.py", line 410, in _init_handles
    raise OutOfResources(self.metadata.shared, max_shared, "shared memory")
```
    
Notes
1. the benchmark has several auto tuning configurations  (including the one above causing compilation failure). The auto tuner hides the functional failure, and ends up choosing a configuration that allows the benchmark to compile. The configuration chosen has block sizes that are too small and consequently the achievable performance is also significantly lower than that of a GEMM without any postOp operations.
2. I believe that fixing the functional problem would "unlock" performance for this benchmark. For a small block size ('BLOCK_SIZE_M': 64, 'BLOCK_SIZE_N': 128, 'BLOCK_SIZE_K': 32,) the performance is just a bit less than a GEMM kernel with no postOp:

**Performance of GEMM (no postOp)**
```
     B       M       K       N  Triton-GB/s  Triton-GB/s-min  Triton-GB/s-max  Triton-TFlops  Triton-TFlops-min  Triton-TFlops-max  Triton-CV
0  1.0  4096.0  4096.0  4096.0    61.981733        61.977154        61.986312      63.469294          63.464605          63.473983   0.000104
```
**Performance of GEMM + postOp (add matrix to result)**
```
     B       M       K       N  Triton-GB/s  Triton-GB/s-min  Triton-GB/s-max  Triton-TFlops  Triton-TFlops-min  Triton-TFlops-max  Triton-CV
0  1.0  4096.0  4096.0  4096.0    54.684536        54.450264        54.920832      55.996965           55.75707          56.238932   0.006085
```

### PR DESCRIPTION
**Pull request by etiotto**
Fixes #2378

### CODE REVIEW

**Comment by etiotto** [comment]
Performance before this PR:
image

Performance with this PR improves significantly (e.g. 8Kx8Kx8K from ~66TFlops to ~215TFlops):
image

**Review by alexbaden — APPROVED** [review]

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp [hunk 1]
```diff
@@ -738,11 +748,11 @@ class TritonIntelGPURewriteTensorPointerPass
...
     LLVM_DEBUG({
       if (valueToRemove.empty())
-        llvm::dbgs() << "No tensor pointer to remove\n";
+        LDBG("No tensor pointer to remove");
       else {
-        llvm::dbgs() << "Values to remove: \n";
+        LDBG("Values to remove: ");
         for (auto val : valueToRemove)
-          llvm::dbgs() << val << "\n";
+          LDBG(val);
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:755]: Does LDBG inside LLVM_DEBUG work?

Comment (whitneywhtsang) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:755]: should we change `llvm::dbgs()` to `DBGS` instead?

**Review by whitneywhtsang — APPROVED** [review]

File: third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp (hunk 1 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUTransforms/RewriteTensorPointer.cpp:755]: good catch, fixing

**Pull request merged**